ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -484,35 +613,66 @@ private DeleteResult run_delete_v2(String accountName, String containerName, Str
         }
         blobId = Base64.encodeBase64URLSafeString(resultSet.getBytes(1));
         version = resultSet.getLong(2);
-        Timestamp deletionTime = resultSet.getTimestamp(3);
-        currentTime = resultSet.getTimestamp(4);
-        alreadyDeleted = (deletionTime != null && currentTime.after(deletionTime));
+        Timestamp originalDeletionTime = resultSet.getTimestamp(3);
+        currentDeleteTime = resultSet.getTimestamp(4);
+        alreadyDeleted = (originalDeletionTime != null && currentDeleteTime.after(originalDeletionTime));
       }
     }
     // only need to issue an update statement if the row was not already marked as deleted.
     if (!alreadyDeleted) {
-      try (PreparedStatement statement = connection.prepareStatement(SOFT_DELETE_QUERY_V2)) {
-        // use the current time
-        statement.setTimestamp(1, currentTime);
-        statement.setInt(2, accountId);
-        statement.setInt(3, containerId);
-        statement.setString(4, blobName);
-        statement.setLong(5, version);
-        statement.executeUpdate();
-      }
+      applySoftDelete(accountId, containerId, blobName, version, currentDeleteTime, connection);
     }
     return new DeleteResult(blobId, alreadyDeleted);
   }
 
+  private List<StaleNamedResult> runPullStaleBlobs(final Connection connection) throws Exception {
+    try (PreparedStatement statement = connection.prepareStatement(GET_STALE_QUERY)) {
+      Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+      calendar.add(Calendar.DATE, -config.staleMaxDays);
```

Comment:
Calendar.DAY_OF_MONTH?

Reply:
I see DAY_OF_MONTH is the same thing as DATE. And I feel 'DAY_OF_MONTH' is a bit confusion here since we really want a unit of DAYS. Not which day of the month.